core/src/editor.rs

```diff
@@ -124,7 +126,14 @@ impl<'a> MapManager<'a> {
         transform.translation.y = position.y;
     }
     fn delete_element(&mut self, entity: Entity) {
-        self.entities.kill(entity);
+        if let Some(fish_school) = self.fish_school.get_mut(entity) {
```

Comment:
Awesome!

Can you move the kill callback to it's own component, such as `ElementKillCallback`?

That will let us check whether or not the element being deleted has a kill callback, without making it specific to the fish school.

That way, all we have to do to make this work for, snails, crabs, items, players, etc. will be to add the `ElementKillCallback` component to their elements.

We should probably put `ElementKillCallback` in the `elements.rs` module, and it'd probably be good to give it a `new()` method that takes a generic type implementing `IntoSystem` ( [see here](https://github.com/fishfolk/jumpy/blob/e456f2f88ea44ed3be0c2330ca08b857a23338b3/core/src/item.rs#L197) for how to do the generic arguments ) so that there's less boilerplate when creating an instance of the component.